### ISSUE TITLE
v3.0.10: Thunderbird displays message after deleting

### ISSUE THREAD

**Issue by fthx** [ProtonMail/proton-bridge#334]
Hi,

In v3.0.10 I get this strange behavior:
1- select a message
2- press Delete
3- message is no more displayed
4- then the message reappears
5- and finally is no more displayed after maybe two seconds

Not happens in 2.4.8.

**Comment by LBeernaertProton** [comment]
Hey @fthx, could you provide us with the version you are using and some log files of this incident? 

Note that Thunderbird is know to defer message removal until a later point in time or when you close the application.

**Comment by fthx** [comment]
Nope, I do use only TB (102.7.1 snap, Ubuntu 22.04).
There's nothing in Ubuntu logs.
I don't know if there are bridge's logs to show.

**Comment by LBeernaertProton** [comment]
You can find the logs for the bridge under `$HOME/.local/share/protonmail/bridge-v3/logs/`

**Comment by fthx** [comment]
Bridge auto-stared, I click on app launcher:
```
INFO[Jan 31 09:38:30.605] bridge-gui starting
INFO[Jan 31 09:38:30.605] Using Qt 6.3.2
ERRO[Jan 31 09:38:30.605] Instance already exists /home/fthx/.cache/protonmail/bridge-v3/bridge-v3-gui.lock (PID : 3381 - Host : legion-fthx - App : bridge-gui)
```

**Comment by fthx** [comment]
Nothing in logs during this issue.
Please note that another mail server I do use does not show such a behavior, with same TB 107 on the same machine.

**Comment by vmivanov** [comment]
I can confirm the same issue with Thunderbird v102.6.1 and Bridge v3.0.12. Nothing suspicious in the logs or in CLI output. Similar to OP this does not happen on other IMAP servers and, as far as I can tell, this issue is specific to Bridge v3 series and does not occur with Bridge v2. 

```
time="Feb  1 11:38:06.064" level=info msg="200 OK: PUT https://mail-api.proton.me/mail/v4/messages/label" userID="<removed>"
time="Feb  1 11:38:06.526" level=info msg="200 OK: GET https://mail-api.proton.me/core/v4/events/<removed>" userID="<removed>"
time="Feb  1 11:38:06.527" level=info msg="Received new API event" new="Event <removed>: messages: created=0, updated=1, deleted=0" old="<removed>" userID="<removed>"
time="Feb  1 11:38:06.527" level=info msg="Handling message updated event" messageID="<removed>" subject="******** (11eeb7e4)" userID="<removed>"
```

Notes:
- I've configured Thunderbird's Delete action to "Move to folder: Deleted" (same for other IMAP servers I have added). The default, I believe, is "Mark as Deleted". This is likely why in the log trace above it shows `updated=1; deleted=0`.

Further observations:
- The issue can also be reproduced by manually moving the message to the "Deleted" folder;
-  However, if the message were to be manually moved to custom folder, the issue is _not_ observed and the message does _not_ briefly reappear.

The above observations make me think it may have to do with the special status of the "Deleted" system folder and how messages are synced when moved there.

Hope this helps.

**Comment by LBeernaertProton** [comment]
Hey @fthx @vmivanov I have confirmed the behavior you described. That being said, shortly afterwards the message does end up in the Trash folder. Do you also experience this or does the message remain in the original location forever?

**Comment by vmivanov** [comment]
Yes, that's correct, once the message disappears for the second time, it ultimately stays in the Deleted/Trash folder and is gone from its original folder.

So the overall end result is correct, though the experience can be frustrating when working through messages in the list (e.g. with up/down keyboard arrows) as the deleted message is selectable for the short period of time that it reappears .

**Comment by fthx** [comment]
Yes, quite annoying when deleting many messages.

**Comment by LBeernaertProton** [comment]
Thanks or the information, we created an internal ticket and working to address the issue.

**Comment by AwlsomeAlex** [comment]
No longer experiencing this with v3.0.14 (but did in 3.0.10 and 3.0.12).

**Comment by vmivanov** [comment]
Can confirm above comment for myself too - v3.0.14 appears to have resolved this issue. I presume the buggy behaviour was fixed as a result of:
> GODT-2323: Fix Expunge not issued for move

**Comment by fthx** [comment]
After some test time, completely solved for me too, 3.0.14+.
